test/controllers/csv_controller_test.rb

```diff
@@ -0,0 +1,198 @@
+require_relative '../test_helper'
+
+describe CsvController do
```

Comment:
missing permission tests ... either structure like this (lowest group first, then test as soon as it is allowed) ... or as_a_viewer/dev/admin per action

```
as_a_viewer do
  index should be forbidden
  post should be forbidden
  status should return x
end

as_a_deployer do
  index should be forbidden
  post should create an export
end

as_a_admin do
  index should return xyz
end
```